sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -131,13 +134,101 @@ object ExtractEquiJoinKeys extends Logging with PredicateHelper {
 
       if (joinKeys.nonEmpty) {
         val (leftKeys, rightKeys) = joinKeys.unzip
-        logDebug(s"leftKeys:$leftKeys | rightKeys:$rightKeys")
-        Some((joinType, leftKeys, rightKeys, otherPredicates.reduceOption(And), left, right))
+
+        // Find any simple range expressions between two columns
+        // (and involving only those two columns)
+        // of the two tables being joined,
+        // which are not used in the equijoin expressions,
+        // and which can be used for secondary sort optimizations.
+        val rangePreds: mutable.Set[Expression] = mutable.Set.empty
```

Comment:
why do we need this? Can't we just use `rangeConditions`?

Reply:
I see that you are using it later but can't you use `rangeConditions` instead? they seem duplicates...they contain the same data IIUC

Reply:
`rangeConditions` contain "vice-versa" conditions in case left and right plans need to be switched.

Reply:
I see... maybe we can make this more clear in the comments then, what do you think?

Reply:
Yes, will do